### ISSUE TITLE
Do vLLM support `input_embeds` as input while using LLama?

### ISSUE THREAD

**Issue by OswaldoBornemann** [vllm-project/vllm#8323]
Can we directly pass the input_embeds to the generate function? Just like the following used in the pytorch transformers

```
generated_ids = self.model.generate(
            inputs_embeds=input_token_embedding,
            do_sample=True,
            max_length=max_length,
            pad_token_id=self.pad_token_id,
            eos_token_id=self.eos_target_id,
            temperature=temperature,
            top_k=top_k,
            top_p=top_p,
            repetition_penalty=repeat_penalty,
            min_new_tokens=50,
        )
```

**Comment by DarkLight1337** [comment]
The current code doesn't support that. You can work around this by leveraging the multi-modal support in vLLM which lets you directly pass multi-modal embeddings to the language backbone. In particular, you can replace multi-modal embeddings with text embeddings, and modify the existing code to accept text embeddings as "multi-modal inputs".

**Comment by DarkLight1337** [comment]
Also see #416

**Comment by OswaldoBornemann** [comment]
Thanks for your amazing and fast response. But in my case, I only have the `input_embeds`, and I don't have any `prompt`. My case is an audio generation situation.

**Comment by OswaldoBornemann** [comment]
Something like below:

```
llm.generate({
    'multi_modal_data': {"image": input_token_embedding}
})
```

**Comment by DarkLight1337** [comment]
You can try passing in a prompt filled with `<image>` tokens (the number equal to the size of your embedding) to a vision-language model and pass your text embeddings as you would do for image embeddings.

**Comment by OswaldoBornemann** [comment]
Thank you @DarkLight1337 . I see. But if that is so, I think it's quite strange. Because it means that i need to include additional modules that I don't require at all.

**Comment by DarkLight1337** [comment]
Yeah, that's why I said in the first comment that it's not really supported yet.

**Comment by OswaldoBornemann** [comment]
What about this one? https://github.com/vllm-project/vllm/pull/6869 @DarkLight1337

**Comment by DarkLight1337** [comment]
Interesting, I completely missed this PR.

**Comment by OswaldoBornemann** [comment]
Thank you. Please take a look if you have time.

**Comment by OswaldoBornemann** [comment]
@DarkLight1337 Hi, is there any way to directory install `VLLM` under this commit? Thanks.

**Comment by OswaldoBornemann** [comment]
Thank you.

**Comment by OswaldoBornemann** [comment]
@DarkLight1337  Hi, have you tried this https://github.com/vllm-project/vllm/pull/6869?

**Comment by DarkLight1337** [comment]
No, I only looked at the code.

**Comment by zengrh3** [comment]
@DarkLight1337 Hi, I noticed lots of progress in adding `input_embeds` in vLLM. So is that ready or not?

**Comment by DarkLight1337** [comment]
The PR is not ready for merge yet. I need to merge a couple of preliminary PRs first so others can better review it.

**Comment by fzyzcjy** [comment]
Hi, is there any updates? Thanks!
